src/arith/iter_affine_map.cc

```diff
@@ -351,8 +374,18 @@ class IterMapRewriter : public ExprMutator {
     std::vector<IterSplitExpr> iters;
     iters.reserve(expr->args.size());
     // canonicalize the expression
+    // find the base scale first
+    PrimExpr base_scale(nullptr);
+    for (const auto& arg : expr->args) {
```

Comment:
We can save one iteration of the for loop below after we find the base scale.

Let us simply remember the index and find the base_index. in args. start with base_index == -1

Then mark the that index as visited. So the next for loop can start with i = 1